src/main-process/atom-application.coffee

```diff
@@ -35,9 +36,14 @@ class AtomApplication
     unless options.socketPath?
       if process.platform is 'win32'
         userNameSafe = new Buffer(process.env.USERNAME).toString('base64')
-        options.socketPath = "\\\\.\\pipe\\atom-#{options.version}-#{userNameSafe}-#{process.arch}-sock"
+        # To have a separate session per ATOM_HOME directory, we append the first
+        # few characters of the the hash of ATOM_HOME to the socket file name.
+        atomHomeUnique = path.resolve(process.env.ATOM_HOME)
+        hash = crypto.createHash('sha1').update(atomHomeUnique)
+        atomHomeDigest = hash.digest('hex').substring(0, 12)
+        options.socketPath = "\\\\.\\pipe\\atom-#{options.version}-#{userNameSafe}-#{process.arch}-#{atomHomeDigest}-sock"
       else
-        options.socketPath = path.join(os.tmpdir(), "atom-#{options.version}-#{process.env.USER}.sock")
+        options.socketPath = path.join(process.env.ATOM_HOME, "atom-#{options.version}-#{process.env.USER}.sock")
```

Comment:
@mortenpi Hi- was just taking a look at this, I don't think you want to put sockets in atom home. In some systems `socket` syscall might be denied by selinux / apparmor / etc, so I think it may be better to use the file name to ensure a unique socket while keeping determinism in place of a folder. I.E. Make this small adjustment:

```coffeescript
    # To have a separate session per ATOM_HOME directory, we append the first
    # few characters of the the hash of ATOM_HOME to the socket file name.
    atomHomeUnique = path.resolve(process.env.ATOM_HOME)
    hash = crypto.createHash('sha1').update(atomHomeUnique)
    atomHomeDigest = hash.digest('hex').substring(0, 12)
    unless options.socketPath?
      if process.platform is 'win32'
        userNameSafe = new Buffer(process.env.USERNAME).toString('base64')
        options.socketPath = "\\\\.\\pipe\\atom-#{options.version}-#{userNameSafe}-#{process.arch}-#{atomHomeDigest}-sock"
      else
        options.socketPath = path.join(os.tmpdir(), "atom-#{options.version}-#{process.env.USER}-#{atomHomeDigest}.sock")
```

Reply:
Thanks for the feedback! I was wondering whether there might be a reason like that to prefer `/tmp` over `/home`. I restored `os.tmpdir()` as you suggested.